src/components/ResponsiveGrid/ResponsiveGrid.tsx

```diff
@@ -0,0 +1,187 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'react-peek/prop-types';
+import { lucidClassNames } from '../../util/style-helpers';
+import {
+	StandardProps,
+	findTypes,
+	omitProps,
+} from '../../util/component-types';
+import Resizer from '../Resizer/Resizer';
+
+const cx = lucidClassNames.bind('&-ResponsiveGrid');
+
+const { string, number, arrayOf } = PropTypes;
+
+interface IResponsiveGridCellProps extends StandardProps {}
+
+const Cell = (_props: IResponsiveGridCellProps): null => null;
+Cell.displayName = 'ResponsiveGrid.Cell';
+Cell.peek = {
+	description: `
+		Renders a \`<span>\` as the grid cell
+	`,
+};
+
+interface IResponsiveGridWrapperProps
+	extends StandardProps,
+		React.DetailedHTMLProps<
+			React.HTMLAttributes<HTMLDivElement>,
+			HTMLDivElement
+		> {
+	/**
+	 * Break points for the grid to switch column layouts. Break points should be provided
+	 * in ascending order. Currently only two break points are used. Example: [960, 1430]
+	 */
+	breakPoints: number[];
+}
+
+interface IResponsiveGridProps extends IResponsiveGridWrapperProps {
+	/**
+	 * Width of the grid. Note: this does not set the width of the grid, and should be
+	 * provided by calculating the width of the parent element.
+	 */
+	width: number;
+}
+
+const defaultProps = {
+	breakPoints: [960, 1430],
+};
+
+export class ResponsiveGrid extends React.Component<IResponsiveGridProps> {
+	static displayName = 'ResponsiveGrid';
+
+	static propTypes = {
+		width: number`Width of the grid. Note: this does not set the width of the grid, and should be
+		provided by calculating the width of the parent element.`,
+
+		breakPoints: arrayOf(number)`
+		Break points for the grid to switch column layouts. Break points should be provided 
+		in ascending order. Currently only two break points are used. Example: [960, 1430]`,
+
+		className: string`
+			Appended to the component-specific class names set on the root elements.
+		`,
+	};
+
+	getColumnLayout = (numberOfColumns: number) => {
+		const cellProps = _.map(
+			findTypes(this.props, ResponsiveGridWrapper.Cell),
+			'props'
+		);
+
+		const columns = _.reduce(
+			/* We use ts-ignore below since the typescript compiler pattern matches this _.map
+			 to an array of booleans */
+			_.map(cellProps, (props: IResponsiveGridCellProps, key: number) => (
+				<article key={key} className={cx('&-Cell')}>
+					{props.children}
+				</article>
+			)),
+			(columns: React.ReactElement[][], cell, idx: number) => {
+				//@ts-ignore
+				columns[idx % numberOfColumns].push(cell);
+				return columns;
+			},
+			_.times(numberOfColumns, () => [])
+		);
+
+		return (
+			<div
+				className={cx('&', {
+					'&-one-column': numberOfColumns === 1,
+				})}
+			>
+				{_.map(columns, (col, key) => {
+					return (
+						<div key={key} className={cx('&-Column')}>
```

Comment:
Should we declare the `key` as something other than the index of the column, since this has the possibility of being re-rendered if a breakpoint changes? A better `key` value isn't immediately popping out at me though.

Reply:
Yeah, I didn't see any options for key here other than the index. Although different groups of columns are rendered as the width changes, the ordering of columns within a group never changes once rendered, so I think it is safe to use the index as a key here.
https://reactjs.org/docs/lists-and-keys.html

Reply:
Ah right, it's re-ordering that is the problematic case, not re-rendering the same list. Agreed on the index issue, thank you!